plugins/wasi_nn/neuralspeed.cpp

```diff
@@ -0,0 +1,338 @@
+      Env.NNGraph.pop_back();
+      return ErrNo::InvalidArgument;
+    }
+    TempFile << BinModel;
+    TempFile.close();
+    if (GraphRef.EnableDebugLog) {
+      spdlog::info(
+          "[WASI-NN][Debug] Neural speed: Write model into a tmpfile...Done"sv);
+    }
+  }
+  if (GraphRef.EnableDebugLog) {
+    spdlog::info(
+        "[WASI-NN][Debug] Neural speed: Finished handling model path."sv);
+  }
+
+  // Create Model class
+  if (!Py_IsInitialized()) {
+    Py_Initialize();
+  }
+  GraphRef.NeuralSpeedModule =
+      PyImport_Import(PyUnicode_FromString("neural_speed"));
+  if (GraphRef.NeuralSpeedModule == nullptr) {
+    PyErr_Print();
+    spdlog::error(
+        "[WASI-NN] neural speed backend: Can not find neural speed library."sv);
+    return WASINN::ErrNo::RuntimeError;
+  }
+  GraphRef.ModelClass =
+      PyObject_GetAttrString(GraphRef.NeuralSpeedModule, "Model");
+  if (GraphRef.ModelClass == nullptr ||
+      !PyCallable_Check(GraphRef.ModelClass)) {
+    spdlog::error(
+        "[WASI-NN] neural speed backend: Can not find Model class in neural speed."sv);
+    Py_XDECREF(GraphRef.Model);
+    return WASINN::ErrNo::RuntimeError;
+  }
+  GraphRef.Model = PyObject_CallObject(GraphRef.ModelClass, NULL);
+  if (GraphRef.Model == nullptr) {
+    spdlog::error("[WASI-NN] neural speed backend: Load model error."sv);
+    Py_XDECREF(GraphRef.ModelClass);
+    Py_XDECREF(GraphRef.NeuralSpeedModule);
+    return WASINN::ErrNo::InvalidArgument;
+  }
+  PyObject *LoadResult =
+      PyObject_CallMethod(GraphRef.Model, "init_from_bin", "(ss)",
+                          GraphRef.model_type.c_str(), ModelFilePath.c_str());
+  if (LoadResult == nullptr) {
+    spdlog::error("[WASI-NN] neural speed backend: Load model error."sv);
+    Py_XDECREF(GraphRef.Model);
+    Py_XDECREF(GraphRef.ModelClass);
+    Py_XDECREF(GraphRef.NeuralSpeedModule);
+    return WASINN::ErrNo::InvalidArgument;
+  }
+  Py_XDECREF(LoadResult);
+  GraphRef.LoadTime = std::chrono::duration_cast<std::chrono::milliseconds>(
+                          std::chrono::steady_clock::now() - StartTime)
+                          .count();
+  return WASINN::ErrNo::Success;
+}
+
+Expect<WASINN::ErrNo> initExecCtx(WasiNNEnvironment &Env, uint32_t GraphId,
+                                  uint32_t &ContextId) noexcept {
+  if (!Py_IsInitialized()) {
+    spdlog::info(
+        "[WASI-NN][Error] Neural speed backend: Model has been realse, please reload it."sv);
+    return WASINN::ErrNo::RuntimeError;
+  }
+  Env.NNContext.emplace_back(GraphId, Env.NNGraph[GraphId]);
+  ContextId = Env.NNContext.size() - 1;
+  return ErrNo::Success;
+}
+
+Expect<WASINN::ErrNo> setInput(WasiNNEnvironment &Env, uint32_t ContextId,
+                               uint32_t, const TensorData &Tensor) noexcept {
+  auto &CxtRef = Env.NNContext[ContextId].get<Context>();
+  auto &GraphRef = Env.NNGraph[CxtRef.GraphId].get<Graph>();
+  if (!Py_IsInitialized()) {
+    spdlog::info(
```

Comment:
We should use `spdlog::error` here.

Reply:
I have changed. Please check, thank you.